samples/subsys/audio/sof/prj.conf

```diff
@@ -5,6 +5,13 @@ CONFIG_BUILD_OUTPUT_BIN=n
 # Requires heap_info() be implemented, but no Zephyr wrapper
 CONFIG_DEBUG_MEMORY_USAGE_SCAN=n
 
+# Ensure -O2 (or similar) is used for optimization
+CONFIG_SPEED_OPTIMIZATIONS=y
+
+# Zephyr top-level default of -fno-strict-overflow is
+# not a good fit for audio DSP code
```

Comment:
IMHO you can be stronger here: "-fno-strict-overflow disables important optimizations on which almost all non-unrollable loops rely and is a poor choice for a default."  But maybe that's just me picking fights.

Reply:
everybody seems to agree, that this is a good optimisation, but TBH it seems less obvious to me. As a coder I'm well aware of integer overflows, so, if I write
```
int i = INT_MIN;
return i - 10 < 0;
```
I **expect** that to underflow, the result to become positive and the condition to return false. I understand that the compiler is trying to be smart here, since it knows, that `ì` is negative, negative minus positive makes negative, so the result is `true`. But in a less obvious form the result is impossible to predict for the compiler, but my expectation doesn't change? Therefore that flag seems quite risky to me to set globally. If we have some highly-optimised consciously manually verified not to overflow code, we could enable that flag for that code, but enabling it globally seems risky to me. Or is it me having wrong expectations and we should really never rely on overflow comparisons? But can we really be sure that those never occur?

Reply:
Did you @lyakh check the https://www.airs.com/blog/archives/120 ? While you expect that to happen, C99 and C++98 state it is in fact undefined behavior and correct code should not rely on it. no-strict-overflow essentially tells the compiler that despite what the spec says, you cannot assume integer-overflow is undefined. And this disabled a category of optimizations.

With the XCC compilers this is worse, as this also affects code that is only using unsigned ints. I don't know why it does it, but that is what I observed and can reproduce, so we'll have to live with it. In short, if we compile with "no-strict-overflow", we'll get slower code without really means to fix it (other than handwriting as inline assembly, which given the big existing codebase in C, is not an option).